spec/record_serialization_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'attr_json/model'
+RSpec.describe "AttrJson::Record serialization" do
+
+  class Product
+    include AttrJson::Model
+    attr_json :name, :string, default: "A Product"
+    attr_json :cost, :string, default: "$2.25"
+    attr_json :material, :string, default: "silk"
+    attr_json :color, :string, default: "green"
+  end
+
+  class Brand
+    include AttrJson::Model
+
+    def self.build_test
+      new(products: [Product.new] )
+    end
+
+    attr_json :name, :string, default: "Some Brand"
+    attr_json :products, Product.to_type, array: true, default: []
+  end
+
+  class Department
+    include AttrJson::Model
+
+    def self.build_test
+      new(brands: [Brand.build_test]  )
+    end
+
+    attr_json :name, :string, default: "Some Department"
+    attr_json :brands, Brand.to_type, array: true, default: []
+  end
+
+
+  class Store < ActiveRecord::Base
+    include AttrJson::Record
+
+    self.table_name = "products"
+
+    attr_json :departments, Department.to_type, container_attribute: :other_attributes, array: true, default: []
+  end
+
+
+  let(:unsaved_store) { Store.new(departments: Array.new(3) { Department.build_test }) }
+
+  describe "#as_json" do
+
+    # This seems to pass, everything is properly json-ified
+    it "produces json-value-only-hash" do
+      expect(only_json_values?( unsaved_store.as_json )).to be true
+    end
+  end
+
+  describe "#serializable_hash" do
+
+    # This does not pass, values are still ruby objects
+    it "produces json-value-only-hash" do
+      expect(only_json_values?( unsaved_store.serializable_hash )).to be true
+    end
```

Comment:
I don't think this should be *expected* to pass. Using standard rails features, serializable_hash sometimes has ruby value objects in it -- such as a `BigDecimal` for a `decimal` column.  It seems wrong to expect `serializable_hash` to convert to json-friendly objects for attr_json, since it doesn't for built-in stuff like `BigDecimal`.  

It seems like that is in fact the difference between `serializable_hash` and `as_json`, why they are two different methods -- it's `as_json` that is expected to return a Hash with only json-compatible classes in it (hashes, arrays, an json-defined primitives).